### PR TITLE
dev-vcs/tortoisehg: 6.1.2 version bump

### PR DESCRIPTION
**Pull request by band-a-prend**
Among other changes after thg-6.1 release the license was updated to GPL-2+

### CODE REVIEW

File: dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{8..10} )
+
+DISTUTILS_USE_SETUPTOOLS=no
```
Comment (arthurzam) [inline dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild:8]: Please use the new `PEP517` mode, docs [here](https://projects.gentoo.org/python/guide/distutils.html).

File: dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	# Remove file that collides with >=mercurial-4.0 (bug #599266).
+	rm "${S}"/hgext3rd/__init__.py || die "can't remove /hgext3rd/__init__.py"
+
+	sed -i -e 's:share/doc/tortoisehg:share/doc/'"${PF}"':' setup.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_test() {
+	${EPYTHON} tests/run-tests.py -m 'not largefiles' --doctest-modules tests || die
+	${EPYTHON} tests/run-tests.py -m largefiles tests || die
```
Comment (arthurzam) [inline dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild:60]: please append `EPYTHON` version for each `die`, so we know with which python it failed if it does.
something like:
```suggestion
	${EPYTHON} tests/run-tests.py -m 'not largefiles' --doctest-modules tests || die "Tests failed with ${EPYTHON}"
	${EPYTHON} tests/run-tests.py -m largefiles tests || die "Tests failed with ${EPYTHON}"
```

File: dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	${HG_DEPEND}
+	dev-python/iniparse[${PYTHON_USEDEP}]
+	dev-python/pygments[${PYTHON_USEDEP}]
+	dev-python/PyQt5[network,svg,${PYTHON_USEDEP}]
+	>=dev-python/qscintilla-python-2.9.4[qt5(+),${PYTHON_USEDEP}]
+"
+DEPEND="
```
Comment (arthurzam) [inline dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild:40]: For most of python cases, you want to use `BDEPEND`:
```suggestion
BDEPEND="
```

**Review by arthurzam — CHANGES_REQUESTED** [review]
Some small changes I saw while doing a short review 😃

File: dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild (hunk 1 above)
Comment (band-a-prend) [reply dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild]: Thank you for review!

Replaced with `DISTUTILS_USE_PEP517=setuptools`.

File: dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild (hunk 2 above)
Comment (band-a-prend) [reply dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild]: Done.
Additionally option `--disable-pytest-warnings` is added for more clean tests output without "deprecation warnings".

File: dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild (hunk 3 above)
Comment (band-a-prend) [reply dev-vcs/tortoisehg/tortoisehg-6.1.2.ebuild]: Done.

**Pull request closed without merging**